src/library/crate/cratefeature.cpp

```diff
@@ -0,0 +1,915 @@
+
+    m_pAutoDjTrackSourceAction = std::make_unique<QAction>(tr("Auto DJ Track Source"),this);
+    m_pAutoDjTrackSourceAction->setCheckable(true);
+    connect(m_pAutoDjTrackSourceAction.get(), SIGNAL(changed()),
+            this, SLOT(slotAutoDjTrackSourceChanged()));
+}
+
+void CrateFeature::connectLibrary(Library* pLibrary) {
+    connect(pLibrary, SIGNAL(trackSelected(TrackPointer)),
+            this, SLOT(slotTrackSelected(TrackPointer)));
+    connect(pLibrary, SIGNAL(switchToView(const QString&)),
+            this, SLOT(slotResetSelectedTrack()));
+}
+
+void CrateFeature::connectTrackCollection() {
+    connect(m_pTrackCollection, SIGNAL(crateInserted(CrateId)),
+            this, SLOT(slotCrateTableChanged(CrateId)));
+    connect(m_pTrackCollection, SIGNAL(crateUpdated(CrateId)),
+            this, SLOT(slotCrateTableChanged(CrateId)));
+    connect(m_pTrackCollection, SIGNAL(crateDeleted(CrateId)),
+            this, SLOT(slotCrateTableChanged(CrateId)));
+    connect(m_pTrackCollection, SIGNAL(crateTracksChanged(CrateId, QList<TrackId>, QList<TrackId>)),
+            this, SLOT(slotCrateContentChanged(CrateId)));
+    connect(m_pTrackCollection, SIGNAL(crateSummaryChanged(QSet<CrateId>)),
+            this, SLOT(slotUpdateCrateLabels(QSet<CrateId>)));
+}
+
+QVariant CrateFeature::title() {
+    return tr("Crates");
+}
+
+QIcon CrateFeature::getIcon() {
+    return m_cratesIcon;
+}
+
+QString CrateFeature::formatRootViewHtml() const {
+    QString cratesTitle = tr("Crates");
+    QString cratesSummary = tr("Crates are a great way to help organize the music you want to DJ with.");
+    QString cratesSummary2 = tr("Make a crate for your next gig, for your favorite electrohouse tracks, or for your most requested songs.");
+    QString cratesSummary3 = tr("Crates let you organize your music however you'd like!");
+
+    QString html;
+    QString createCrateLink = tr("Create New Crate");
+    html.append(QString("<h2>%1</h2>").arg(cratesTitle));
+    html.append("<table border=\"0\" cellpadding=\"5\"><tr><td>");
+    html.append(QString("<p>%1</p>").arg(cratesSummary));
+    html.append(QString("<p>%1</p>").arg(cratesSummary2));
+    html.append(QString("<p>%1</p>").arg(cratesSummary3));
+    html.append("</td><td rowspan=\"2\">");
+    html.append("<img src=\"qrc:/images/library/crates_art.png\">");
+    html.append("</td></tr>");
+    html.append(QString("<tr><td><a href=\"create\">%1</a>")
+                .arg(createCrateLink));
+    html.append("</td></tr></table>");
+    return html;
+}
+
+std::unique_ptr<TreeItem> CrateFeature::newTreeItem(
+        const CrateSummary& crateSummary,
+        TrackId selectedTrackId) {
+    auto pTreeItem = std::make_unique<TreeItem>(
+            this, formatLabel(crateSummary), crateSummary.getId().toVariant());
+    pTreeItem->setIcon(
+            crateSummary.isLocked() ? m_lockedCrateIcon : QIcon());
+
+    updateTreeItem(pTreeItem.get(), selectedTrackId, std::vector<CrateId>());
```

Comment:
I think this can be removed because the TreeItem is not bold and it will not become bold because of an empty list.